src/rgw/rgw_user.cc

```diff
@@ -1850,6 +1861,12 @@ int RGWUser::check_op(RGWUserAdminOpState& op_state, std::string *err_msg)
     return -EINVAL;
   }
 
+  if (rgw_is_valid_tenant(op_id.tenant)) {
```

Comment:
we could either return `ERR_INVALID_TENANT_NAME` as returned by the function and reraise in rgw_admin.cc , or abort early in `rgw_rest_user.cc` doing the check.